src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePickerFlyout.iOS.cs

```diff
@@ -119,34 +158,39 @@ private void DatePickerFlyout_Opening(object sender, EventArgs e)
 				this.Dispatcher.RunAsync(Core.CoreDispatcherPriority.Normal, async () =>
 				{
 					await Task.Delay(100);
-					((DatePickerSelector)Content).Date = validDate;
+					_selector.Date = validDate;
 				});
 			}
 		}
 
 		private void DatePickerFlyout_Closed(object sender, EventArgs e)
 		{
-			Date = ((DatePickerSelector)Content).Date;
+			_selector.Cancel();
 		}
 
-		protected override Control CreatePresenter()
+		private void Accept()
 		{
-			_datePickerPresenter = new DatePickerFlyoutPresenter()
-			{
-				Content = Content
-			};
+			_selector.SaveValue();
+			Hide(false);
 
-			AttachAcceptCommand(_datePickerPresenter);
+			DatePicked?.Invoke(this, new DatePickedEventArgs(_selector.Date, Date));
+		}
 
-			return _datePickerPresenter;
+		private void Dismiss()
+		{
+			_selector.Cancel();
+			Hide(false);
 		}
 
-		private void AttachAcceptCommand(IFrameworkElement rootControl)
+		private void AttachFlyoutCommand(string targetButtonName, Action<DatePickerFlyout> action)
 		{
-			var acceptButton = rootControl.FindName("AcceptButton") as Button;
-			if (acceptButton != null && acceptButton.Command == null)
+			if (_presenter.FindName(targetButtonName) is Button button)
 			{
-				acceptButton.Command = new DelegateCommand(Hide);
+				if (button.Command == null)
+				{
+					var self = WeakReferencePool.RentSelfWeakReference(this);
```

Comment:
Is the usage of `WeakReferencePool.RentSelfWeakReference` necessary here?

Reply:
I think it's a good idea, since we're holding a reference to `_presenter` which has the button as a visual descendant, in order to avoid a reference cycle